install_config/persistent_storage/pod_security_context.adoc

```diff
@@ -11,144 +11,406 @@
+fails then the pod will also fail.
+
+SCCs define strategies, such as `RunAsUser`, `supplementalGroups`, and `fsGroup`. These
+strategies are used to help decide if the pod is authorized. Strategy values set to
+_RunAsAny_ are essentially stating that the pod can do what it wants regarding that
+strategy. Authorization is skipped for that strategy and no Openshift default is produced
+based on that strategy. So ids and selinux labels in the resulting container are based on
+*_container_* defaults and not on Openshift policies. Here's a quick summary of _RunAsAny_:
+
+- any id defined in the pod spec (or image) is allowed,
+- absence of an id in the pod spec (and in the image) results in the container assigning an id,
+which is *root* (0) for docker,
+- no selinux labels are defined, so docker will assign a unique (and generally not useful) label.
+
+For these reasons, SCCs with _RunAsAny_ for id related strategies should be protected so
+that ordinary developers do not have access to the SCC.
+
+On the other hand, SCC strategies set to _MustRunAs_ or _MustRunAsRange_ trigger id validation
+(for id related strategies), and cause default values to be supplied by Openshift to the
+container (when those values are not supplied directly in the pod spec or image).
+
+SCCs may define the range of allowed ids (user, groups). If range checking is required
+(e.g., _MustRunAs_) and the allowable range is not defined in the SCC, then the project
+determines the id range. So, projects support ranges of allowable ids; however, unlike SCCs,
+projects do not define strategies, such as `runAsUser`. Allowable ranges are interesting
+not only because they define the boundaries for container IDs, but also because the
+minimum value in the range becomes the *default* value for the id in question. For instance,
+if the SCC id strategy value is _MustRunAs_, and the minimum value of an id range is 100,
+and the id is absent from the pod spec, then 100 is provided as the default for this id.
+
+As part of pod authorization, the SCCs available to a pod are examined (roughly, in priority
+order followed by most restrictive) to best match the requests of the pod. A SCC's strategy's
+type of _RunAsAny_ is less restrictive; wheras a type of _MustRunAs_ is more restrictive. All
+of these strategies are evaluated. To see which SCC was assigned to a pod use the `oc get pod`
+command, below:
+```
+# oc get pod <pod-name> -o yaml 
+...
+metadata:
+  annotations:
+    openshift.io/scc: nfs-scc <1>
+  name: nfs-pod1 <2>
+  namespace: default <3>
+...
+```
+<1> this is the name of the SCC that the pod used (in this case, a custom SCC)
+<2> this is the name of the pod
+<3> this is the name of the project
+
+*Note:* it can be inobvious which SCC was matched by a pod, so the command above can be very useful
+in understanding the UID, supplemental groups, and selinux re-labeling in a live container.
+
+There are currently six predefined SCCs, seen below:
+```
+# oc get scc
```

Comment:
> perhaps it would be better to not list the predefined SCCs since this will likely change over time. Instead, just point out the SCCs relevant for the discussion below.

+1